specs/core/0_beacon-chain.md

```diff
@@ -1238,8 +1229,8 @@ First, two helper functions:
 def min_empty_validator_index(validators: List[ValidatorRecord],
                               validator_balances: List[int],
                               current_slot: int) -> int:
-    for i, (v, vbal) in enumerate(zip(validators, validator_balances)):
-        if vbal == 0 and v.latest_status_change_slot + ZERO_BALANCE_VALIDATOR_TTL <= current_slot:
+    for i, v in enumerate(validators):
+        if v.withdrawal_slot + ZERO_BALANCE_VALIDATOR_TTL <= current_slot:
```

Comment:
For phases 0 and 1 I'd simply have `min_empty_validator_index` return `None` all the time. (I.e. remove `min_empty_validator_index` and `ZERO_BALANCE_VALIDATOR_TTL` logic.)

Reply:
We need to keep `ZERO_BALANCE_VALIDATOR_TTL` because there's logic constricting deposits that's dependent on that parameter (technically, we could delay that logic until phase 2 as well, but IMO that would cause more harm than good as it would just require us to add more special-case logic).

I'm ok with making `min_empty_validator_index` a stub for now.

Reply:
I made some cleanups here which I'm happy to revert. I removed `min_empty_validator_index` and the associated logic, which removed a decent number of lines of code. I left the explanation message that withdrawn validator indices will be recycled.

I also removed the `ZERO_BALANCE_VALIDATOR_TTL` logic because I felt that that we don't want to unnecessarily commit to somewhat arbitrary parameters and logic. I guess I separate phases 0, 1, 2 quite strictly, with the exception of stubbing data structures. (Note: I personally lean towards @benjaminion (see [here](https://notes.ethereum.org/c/Sk8Zs--CQ/https%3A%2F%2Fbenjaminion.xyz%2Fnewineth2%2F20181228.html)) on not even adding placeholders in phase 0.)

Reply:
Looks good to me!